### DISCUSSION TITLE
Packagerules is ignored when trying to group packages

### DISCUSSION THREAD

**Discussion by davidusken** [renovatebot/renovate#26982] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.159.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### What is your question?

Trying to get started with tuning our Renovate to reduce some noise, first of all by grouping together a few packages that frequently get updated. But for some reason Renovate is not grouping them, I hope someone here can tell me why. The package rules are this:

```
          "packageRules": [
            {
              "matchPackageNames": [
                "prometheus",
                "promtail",
                "loki",
                "mimir-distributed"
              ],
              "groupName": "monitoring packages",
              "groupSlug": "monitoring-packages"
            },
            {
              "matchPackageNames": [
                "renovate"
              ],
              "matchUpdateTypes": [
                "minor", 
                "patch", 
                "digest"
              ],
              "groupName": "renovate packages",
              "groupSlug": "renovate-packages"
            }
          ]
        }
        '
```

Renovate runs fine still, and creates the PRs for each package but not as groups.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
There's nothing wrong with the packageRules you pasted, so it must be something else in your environment or config which you have overlooked. To help you more would require either:
- Full debug logs from a repo run, or
- Reproduction repo (e.g. 2 dependencies, add your packageRules grouping)

**Reply by davidusken** [reply]
Ok good to know! The full config is this:
```

        '
        {
          "platform": "bitbucket-server",
          "endpoint": "https://xxx.xxx.net/bitbucket",
          "autodiscover": false,
          "repositories": [
            "infra/x1",
            "infra/x2",
            "infra/x3"
          ],
          "packageRules": [
            {
              "matchPackageNames": [
                "prometheus",
                "promtail",
                "loki",
                "mimir-distributed"
              ],
              "groupName": "monitoring packages",
              "groupSlug": "monitoring-packages"
            },
            {
              "matchPackageNames": [
                "renovate"
              ],
              "matchUpdateTypes": [
                "minor", 
                "patch", 
                "digest"
              ],
              "groupName": "renovate packages",
              "groupSlug": "renovate-packages"
            }
          ]
        }
        '
```
Deployed with Helm and these are the only env vars changed:
  env:
    RENOVATE_DETECT_HOST_RULES_FROM_ENV: True
    LOG_LEVEL: debug

And the host rules adds these env vars:
```
RENOVATE_USERNAME
RENOVATE_PASSWORD
GITHUB_COM_TOKEN
DOCKER_QUAY_X_NET_USERNAME
DOCKER_QUAY_X_NET_PASSWORD
```

Are you able to see anything here that could cause it? If not I can post debug logs but takes a while to sanitize

**Reply by rarkins** [reply]
I don't see anything wrong there either

**Reply by davidusken** [reply]
Ok thanks for checking! I will try to set up a Reproduction repo. 
One thing I was wondering, do I have to do something with the renovate.json in each repository? I kind of just assumed my Renovate config takes presedence. Currently it only contains this

```
renovate.json

{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json"
}
```

**Reply by rarkins** [reply]
It's fine to have it empty like that

**Comment by davidusken** [comment]
Looks like I got it now, not a bug but my mistake. From what I learned now it seems there must be 2 packages eligible to be added to a group, for the group to be created. And these 2 packages cannot be from different repositories, a package per repository is created.

**Reply by aburke-incomm** [reply]
Would it be possible to provide a link to the reading that allowed you to learn more about this requirement?

**Reply by steffakasid** [reply]
I would also be interested on a link about hat behavior or was it just something you observed.
